### PR TITLE
Us 793 allow multiple actions on modifications

### PR DESCRIPTION
**Pull request by JacquesBorsenberger**

### CODE REVIEW

**Comment by JacquesBorsenberger** [comment]
https://github.com/gridsuite/study-server/pull/181
https://github.com/gridsuite/network-modification-server/pull/84

File: src/components/util/checkbox-list.js [hunk 1]
```diff
@@ -4,57 +4,67 @@
...
+    }, [values, checked, setChecked]);
 
-    const handleToggle = (value) => {
-        const newChecked = new Set(checked);
-        if (!newChecked.delete(value)) {
-            newChecked.add(value);
-        }
-        setChecked(newChecked);
+    const refVals = useRef({ values: onChecked });
+    refVals.current = { values, onChecked };
```
Comment (FranckLecuyer) [inline src/components/util/checkbox-list.js:31]: refVals is already initialized the line before

File: src/components/util/checkbox-list.js [hunk 2]
```diff
@@ -4,57 +4,67 @@
...
+            return oldVals.size > 0
+                ? new Set()
+                : new Set(refVals.current.values);
+        });
+    }, [toggleSelectAll]);
+
+    const handleToggle = useCallback(
+        (value) => {
+            const newChecked = new Set(checked);
+            if (!newChecked.delete(value)) {
```
Comment (FranckLecuyer) [inline src/components/util/checkbox-list.js:45]: why not using Set.has to check if element exists before adding to the Set ?

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]

File: src/components/util/checkbox-list.js (hunk 2 above)
Comment (JacquesBorsenberger) [reply src/components/util/checkbox-list.js:45]: this way there is only one search in the Set

File: src/components/graph/menus/network-modification-node-editor.js [hunk 3]
```diff
@@ -40,8 +43,35 @@ const useStyles = makeStyles((theme) => ({
...
+    },
+    filler: {
+        flexGrow: 1,
+    },
+    dividerTool: {
+        background: theme.palette.primary.main,
+    },
 }));
 
+function isChecked(s1, s2) {
```
[the remark this answers is not in the document]
Comment (JacquesBorsenberger) [reply src/components/graph/menus/network-modification-node-editor.js]: fixed

File: src/utils/rest-api.js [hunk 4]
```diff
@@ -773,8 +773,9 @@ export function deleteModification(studyUuid, node, modificationUuid) {
         encodeURIComponent(studyUuid) +
```
[the remark this answers is not in the document]
Comment (JacquesBorsenberger) [reply src/utils/rest-api.js:773]: fixed

File: src/components/util/checkbox-list.js [hunk 5]
```diff
@@ -4,57 +4,67 @@
...
  * file, You can obtain one at http://mozilla.org/MPL/2.0/.
  */
-import React, { useEffect, useState } from 'react';
+import React, { useCallback, useEffect, useRef, useState } from 'react';
 import List from '@material-ui/core/List';
-import ListItemWithDeleteButton from './list-item-with-delete-button';
 
-const CheckboxList = (props) => {
+const CheckboxList = ({
+    itemRenderer,
```
[the remark this answers is not in the document]
Comment (JacquesBorsenberger) [reply src/components/util/checkbox-list.js:12]: done

**Review by niconoir — APPROVED** [review]

**Review by FranckLecuyer — APPROVED** [review]

**Pull request merged**